From 063ef044260e7101a1976ad810dc613a756bcf8c Mon Sep 17 00:00:00 2001
From: thampan <joby.thampan@intel.com>
Date: Fri, 26 Jun 2020 17:45:41 +0200
Subject: [PATCH] PONRTSYS-7705: Set GSWIP Yellow threshold to a half of green

Setting Yellow threshold to a half of green to don't drop the packet
once marked to yellow color
---
 drivers/net/ethernet/lantiq/switch-api/gsw_defconf.c | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/drivers/net/ethernet/lantiq/switch-api/gsw_defconf.c b/drivers/net/ethernet/lantiq/switch-api/gsw_defconf.c
index 93d62b383c37..a2091ba22ab6 100644
--- a/drivers/net/ethernet/lantiq/switch-api/gsw_defconf.c
+++ b/drivers/net/ethernet/lantiq/switch-api/gsw_defconf.c
@@ -245,9 +245,19 @@ int gsw_qos_def_config(struct core_ops *ops)
 	u32 phy_eg_qmap = BIT(16) | BIT(8) | BIT(0);
 	GSW_QoS_WRED_PortCfg_t sVar;
 	GSW_QoS_WRED_QueueCfg_t qcfg;
+	GSW_QoS_WRED_Cfg_t q_gbl_cfg;
 	GSW_register_t reg;
 	int j;
 
+	/* Setting Yellow threshold to a half of green to don't drop the packet
+	 * once marked to yellow color
+	 */
+	memset(&q_gbl_cfg, 0x00, sizeof(q_gbl_cfg));
+	ops->gsw_qos_ops.QoS_WredCfgGet(ops, &q_gbl_cfg);
+	q_gbl_cfg.nYellow_Min = q_gbl_cfg.nGreen_Min / 2;
+	q_gbl_cfg.nYellow_Max = q_gbl_cfg.nGreen_Max / 2;
+	ops->gsw_qos_ops.QoS_WredCfgSet(ops, &q_gbl_cfg);
+
 	/* Setting Port WRED for all ports */
 	memset(&sVar, 0x00, sizeof(sVar));
 	sVar.nYellow_Min = 240;
